src/scripts/clientUtils.js

```diff
@@ -29,20 +29,43 @@ export async function serverAddress() {
     // if (current) return Promise.resolve(current);
 
     const urls = [];
+    if (window.location.href.indexOf('/web/') !== -1) { // Do we polyfill String.prototype.includes?
+        const split = window.location.href.split('/web/');
+
+        for (let i = split.length - 1; i ; i--) {
+            urls.push(split.slice(0, i).join('/web/'));
+        }
+    }
     urls.push(window.location.origin);
     urls.push(`https://${window.location.hostname}:8920`);
     urls.push(`http://${window.location.hostname}:8096`);
     urls.push(...await webSettings.getServers());
 
+    console.log('URL candidates:', urls);
+
     const promises = urls.map(url => {
-        return fetch(`${url}/System/Info/Public`).then(resp => url).catch(error => {
+        return fetch(`${url}/System/Info/Public`).then(resp => {
+            return {
+                url: url,
+                response: resp
+            };
+        }).catch(error => {
             return Promise.resolve();
         });
     });
 
     return Promise.all(promises).then(responses => {
-        responses = responses.filter(response => response);
-        return responses[0];
+        responses = responses.filter(obj => obj && obj.response.ok);
+        return Promise.all(responses.map(obj => {
+            return {
+                url: obj.url,
+                config: obj.response.json()
+            };
+        }));
+    }).then(configs => {
+        let selection = configs.find(obj => !obj.config.StartupWizardCompleted);
+        if (!selection) selection = configs[0];
+        return Promise.resolve(selection.url);
```

Comment:
```suggestion
        return Promise.resolve(configs.find(obj => !obj.config.StartupWizardCompleted) || configs[0]);
```

Reply:
Why? This changes behavior, and will most likely break everything.

Reply:
I think you forgot to return `.url` there in the suggestion, otherwise it's certainly more compact.

Reply:
`return Promise.resolve((configs.find(obj => !obj.config.StartupWizardCompleted) || configs[0]).url);`

Reply:
```js
let selection = configs.find(obj => !obj.config.StartupWizardCompleted) || configs[0];
return Promise.resolve(selection.url);
```

Reply:
I mean, it's a 2-liner but I think it's worth it for the readability.